src/interpolator/interpolator.c

```diff
@@ -0,0 +1,87 @@
+#include <aero/interpolator/interpolator.h>
+#include <stdlib.h>
+#include <tgmath.h>
+
+typedef struct aero_interpolator_data_elem_t {
+  int from_;
+  int to_;
+  aero_real_t weight_;
+} aero_interpolator_data_elem_t;
+
+struct aero_interpolator_data_t {
```

Comment:
Can we name this something other than `data`? I think it's too easy for people to view interpolation "data" as the data being interpolated and not the weights. Maybe `aero_interpolation_weights_t`?

Also, I wonder if `interp` might be as descriptive and shorter than `interpolator`. My opinions about acronyms tend to evolve pretty continuously these days (I blame the diversity of my clients).

Reply:
I'm ok with `aero_interpolator_weights_t`, although part of me does prefer to follow a common naming scheme: `array_t` with `array_data_t`, `interpolator_t` with `interpolator_data_t`, `foo_t`, with `foo_data_t`, etc.

I am happy to take the flack for long names. Abbreviations are out of control in the science code, and I try wherever I can to adopt Google's C++ style of no abbreviations that you can't find on Wikipedia (except for local variables with limited scope). What do you think?

Reply:
To me, too much emphasis is placed on "consistency" in software engineering naming conventions, in science and in general. I think consistency must give way to clarity and correctness. In this case,  "array data" is clear and unambiguous because an array is a container of data. But "interpolator data" runs the risk of confusing the data the interpolator maintains (which some might even call "metadata") with the data it operates on ("interpolated data", for those of us detail-oriented folk, who are unfortunately in the minority in climate science).

Reply:
Regarding naming, I think the longer names are okay, but again I think we have to pay attention to the crowd we're playing to, so let's keep this in mind. Software engineers are a much more homogeneous group than scientists in their training.

Reply:
I understand the argument about clarity, although I think there is some value in consistency (aside from giving us pedants something to do). If I'm looking at a code base for the first time, patterns in naming (functions, and particularly classes) give helpful clues about patterns in implementation. If we had a number of classes implemented and the `foo_t`/`foo_data_t` naming pattern was evident, I think it would help someone new better understand how best to implement a new class of their own. But, I agree about the specific confusion that `interpolator_data_t` could introduce, so I'm ok with changing it.

For abbreviations, if it were just an aesthetic thing, I wouldn't care as much. But, the incredible number of unintelligible (even by experts) abbreviations is causing real problems in maintaining and extending the code. I think this is one of those cases where we need to encourage (whenever and however possible) scientists to write code for the next person and not just themselves. If they already did this, then `interp` for `interpolator` would be fine to me, but abbreviations are a slippery slope and I think (just based on the current state of science code) that erring on the side of full names is warranted.

Reply:
Fair enough on both points (though there is no argument that can convince me that clarity does not always trump consistency--I've actually seen the latter used as an argument not to have to document anything, which is at best disingenuous and at worst idiotic.) Naming is hard enough without burdening it with making things "obvious" when _there is no such thing in interdisciplinary science._

You are certainly correct that abbreviations are a slippery slope.